bftengine/src/bftengine/PersistentStorageImp.cpp

```diff
@@ -517,8 +531,91 @@ void PersistentStorageImp::setUserDataInTransaction(const void *data, std::size_
   metadataStorage_->writeInBatch(USER_DATA, static_cast<const char *>(data), numberOfBytes);
 }
 
+bool PersistentStorageImp::setReplicaSpecificInfo(uint32_t clientId,
+                                                  uint64_t requestSeqNum,
+                                                  char *rsiData,
+                                                  size_t rsiSize) {
+  // First, find the next available sequence number for this client
+  beginWriteTran();
+  if (rsiLatestIndex.count(clientId) == 0) rsiLatestIndex[clientId] = 0;
+  auto client_index = rsiLatestIndex.at(clientId)++;
+  auto client_circular_index = client_index % numClientBatch_;
+  uint32_t viewChangeMsgsNum = DescriptorOfLastNewView::getViewChangeMsgsNum();
+  uint32_t baseDescNum = clientsDataDescNum + viewChangeMsgsNum + clientId * numClientBatch_;
+  UniquePtrToChar outBuf(new char[replicaSpecificInfoMaxSize + 2 * sizeof(uint64_t) + sizeof(size_t)]);
```

Comment:
Could you use variable names instead of sizeof(uint64_t) and sizeof(size_t)? It is not clear overwise.
You can define constants for those parameters and re-use them. There are a lot of such places below.

Reply:
fixed